In [1]:
import scipy.io as io

# import libs
import numpy as np
import pandas as pd
import os
import scipy.io as io
from sklearn import preprocessing
from scipy.stats import pearsonr
from sklearn.feature_selection import SelectKBest
from sklearn.feature_selection import chi2
from numpy import array
from sklearn.feature_selection import RFE
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.feature_selection import SelectFromModel
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.ensemble import ExtraTreesClassifier

from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, confusion_matrix, classification_report

In [2]:
real_mat_path = '/mnt/hgfs/linuxfile/mat/test_Y.mat'
pred_mat_path = '/mnt/hgfs/linuxfile/mat/test_Y_hat.mat'

In [3]:
test_Y_hat = io.loadmat(pred_mat_path)
test_Y = io.loadmat(real_mat_path)

In [4]:
test_Y_hat = test_Y_hat['label']

In [5]:
test_Y = test_Y['test_Y'].T

In [6]:
print(accuracy_score(test_Y, test_Y_hat))
print(precision_score(test_Y, test_Y_hat,average='micro'))
print(recall_score(test_Y, test_Y_hat,average='micro'))
print(f1_score(test_Y, test_Y_hat,average='micro'))
print(classification_report(test_Y, test_Y_hat))

0.4318297889520272
0.4318297889520272
0.4318297889520272
0.4318297889520272
              precision    recall  f1-score   support

           0       0.93      0.26      0.41     48395
           1       0.49      0.49      0.49     49221
           2       0.18      0.98      0.31     48389
           3       0.99      0.29      0.45     43546
           4       1.00      0.58      0.73     48560
           5       0.89      0.18      0.30     49897
           6       0.27      0.99      0.42     11296
           7       0.77      0.24      0.37     48595
           8       0.50      0.18      0.27     49340
           9       1.00      0.96      0.98     49083
          10       0.39      0.00      0.01     49063

    accuracy                           0.43    495385
   macro avg       0.67      0.47      0.43    495385
weighted avg       0.70      0.43      0.43    495385



0.4318297889520272

0.4318297889520272

0.4318297889520272

0.4318297889520272

            precision    recall  f1-score   support

           0       0.93      0.26      0.41     48395
           1       0.49      0.49      0.49     49221
           2       0.18      0.98      0.31     48389
           3       0.99      0.29      0.45     43546
           4       1.00      0.58      0.73     48560
           5       0.89      0.18      0.30     49897
           6       0.27      0.99      0.42     11296
           7       0.77      0.24      0.37     48595
           8       0.50      0.18      0.27     49340
           9       1.00      0.96      0.98     49083
          10       0.39      0.00      0.01     49063

    accuracy                           0.43    495385
   macro avg       0.67      0.47      0.43    495385
weighted avg       0.70      0.43      0.43    495385

